File: src/main/java/com/muxiu1997/sharewhereiam/util/WaypointBase64.java
package com.muxiu1997.sharewhereiam.util;

import journeymap.client.model.Waypoint;

import java.nio.charset.StandardCharsets;
import java.util.Base64;

public class WaypointBase64 {
    public static String toBase64(Waypoint waypoint) {
        final String waypointJSON = waypoint.toString();
        return Base64.getEncoder().encodeToString(waypointJSON.getBytes(StandardCharsets.UTF_8));
    }

    public static Waypoint fromBase64(String base64) {
        final String waypointJSON = new String(Base64.getDecoder().decode(base64), StandardCharsets.UTF_8);
        return Waypoint.fromString(waypointJSON);
    }
}

File: src/main/kotlin/com/muxiu1997/sharewhereiam/util/WaypointBase64.kt
package com.muxiu1997.sharewhereiam.util

import journeymap.client.model.Waypoint
import java.nio.charset.StandardCharsets
import java.util.*

object WaypointBase64 {
    fun toBase64(waypoint: Waypoint): String {
        val waypointJSON = waypoint.toString()
        return Base64.getEncoder().encodeToString(waypointJSON.toByteArray(StandardCharsets.UTF_8))
    }

    fun fromBase64(base64: String?): Waypoint {
        val waypointJSON = String(Base64.getDecoder().decode(base64), StandardCharsets.UTF_8)
        return Waypoint.fromString(waypointJSON)
    }
}
